akka-distributed-data/src/main/scala/akka/cluster/ddata/ORSet.scala

```diff
@@ -148,6 +207,21 @@ final class ORSet[A] private[akka] (
 
   type T = ORSet[A]
 
+  // Optimization for add/remove followed by merge, which is very common
+  // and merge should just fast forward to the new set
+  private var ancestor: ORSet[A] = null
```

Comment:
yes, unless the `ORSet` is used outside of the `Replicator` infrastructure (and I think you are right in that the worst thing that can happen is that performs a full merge). I'll add comment.